go-controller/pkg/ovn/libovsdbops/model_client.go

```diff
@@ -233,6 +233,11 @@ func (m *ModelClient) buildOps(doWhenFound opModelToOpMapper, doWhenNotFound opM
 				return fmt.Errorf("unexpectedly found multiple results for provided predicate")
 			}
 			hadExistingResults = true
+			// for non-bulk op cases, copy (the one) uuid found to model provided for convenience
+			if opModel.Model != nil && opModel.ModelPredicate != nil && !opModel.BulkOp {
```

Comment:
@astoycos @jcaamano @trozet @dceara this is where we are now populating the uuid in the model

Reply:
I don't think you need to check `!opModel.Bulkop` again above, we would have already exited here if multiple results were found 

Also I might consider writing a new `model client` function rather than adding this complexity to `buildOps` That follows the same sort of logic we use in `func (m *ModelClient) get(opModel *OperationModel) (interface{}, error)` 

So something like 
```
// Unlike m.get this is only called from buildOps so we don't need to copy the Model/Existing Result 
func (m *ModelClient) getConditional (opModel *OperationModel) (error) {
        if err := m.client.WhereCache(opModel.ModelPredicate).List(ctx, opModel.ExistingResult); err != nil {
				return nil, fmt.Errorf("unable to list items for model, err: %v", err)
			}
	uuid := getUUID(opModel.Model)
	if (uuid == "" || IsNamedUUID(uuid)) && !opModel.BulkOp{
		setUUID(opModel.Model, opModel.ExistingResult[0].UUID)
	}
	return copy, nil
}
```

Reply:
hm... not sure if I can do that w/out introducing some reflection-kung-fu. That is so, because `opModel.ExistingResult` is actually an interface{}. So it would require me to do all that 'ugly' stuff
that is currently being done in the [onModels function](https://github.com/ovn-org/ovn-kubernetes/blob/287f819ed764b05f770705339e01e9dadb4f72b6/go-controller/pkg/ovn/libovsdbops/model.go#L226).

 I need to check `!opModel.Bulkop` because there can be multiple uuids in opModel.ExistingResult and it
 would be confusing to know which one we should use. This was the compromise we are doing by adding
 this convenience, I suppose.  @astoycos  @jcaamano what do you think? Does that make sense?

Reply:
> hm... not sure if I can do that w/out introducing some reflection-kung-fu.


I don't think so `if err := m.client.WhereCache(opModel.ModelPredicate).List(ctx, opModel.ExistingResult)` takes care of deciding the type of `opModel.ModelPreedicate`'s type is (we do it in buildops already) 

And Right but in the beginning of the `onModels` function we exit right away if we've already processed an existing result and don't expect another 

```
if hadExistingResults && !opModel.BulkOp {
				return fmt.Errorf("unexpectedly found multiple results for provided predicate")
			}
```
But if you were to break this up and do it before the `onModels` function you would need check it

Reply:
Ack. Please have a look at the changes, as well as the unit test added, @astoycos . Thanks for the feedback!